src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const groupKey = item[key] as string;
```

Comment:
Casting `item[key]` as string assumes that the key will always be a string. This might not be safe if T's properties are not always strings. Consider adding a runtime check to ensure that the key is indeed a string or handle cases where it might not be.